yeat/config/assembly.py

```diff
@@ -55,3 +62,47 @@ def check_canu_required_params(self):
         if self.threads < 4:
             message = f"Canu requires at least 4 avaliable cores; increase '-t' or '--threads' to 4 or more"
             raise AssemblyConfigError(message)
+
+    def check_sample_readtypes_match_assembly_mode(self):
+        for sample in self.samples.values():
+            check = self.is_mode_and_sample_readtypes(sample)
+            self.check_sample_readtypes_match(check, sample)
+
+    def is_mode_and_sample_readtypes(self, sample):
+        if self.mode == "paired" and sample.short_readtype != "paired":
+            return False
+        elif self.mode == "single" and sample.short_readtype != "single":
+            return False
+        elif self.mode == "pacbio" and sample.long_readtype not in PACBIO_READS:
+            return False
+        elif self.mode == "oxford" and sample.long_readtype not in OXFORD_READS:
+            return False
+        elif self.mode == "hybrid" and (
+            sample.short_readtype != "paired" or sample.long_readtype not in LONG_READS
+        ):  # pragma: no cover
+            return False
+        return True
+
+    def check_sample_readtypes_match(self, check, sample):
+        if check == False:
+            message = f"No readtypes in '{sample.label}' match '{self.label}' assembly mode '{self.mode}'"
+            raise AssemblyConfigError(message)
```

Comment:
These three functions ensures that the samples associated with the assembly object has read types matching the assembly's mode.

For example,
Good - assembly mode: "paired" and sample's read type: "paired"
Bad - assembly mode: "paired" and sample's read type: "pacbio-hifi"

Reply:
Really liked this suggestion. Especially for the final "else" statement. I was able to remove `# pragma: no cover` and improve test coverage.